core/chains/evm/log/registrations.go

```diff
@@ -93,13 +104,34 @@ func (r *registrations) addSubscriber(sub *subscriber) (needsResubscribe bool) {
 	return
 }
 
-func (r *registrations) removeSubscriber(sub *subscriber) (needsResubscribe bool) {
+// checkAddSubscriber makes sure we aren't violating any assumptions by adding
+// this subscriber
+func (r *registrations) checkAddSubscriber(sub *subscriber) error {
```

Comment:
Same comment as checkRemoveSubscriber. This function doesn't just check violations, it also starts adding the subscriber.

Reply:
It does not add any subsciber - the maps are only used for invariant checks. I will add an extra comment explaining this.